daemon/mgr/spec_mount.go

```diff
@@ -83,3 +96,78 @@ func setupMounts(ctx context.Context, c *Container, s *specs.Spec) error {
 	}
 	return nil
 }
+
+// generateNetworkMounts will generate network mounts.
+func generateNetworkMounts(c *Container) []specs.Mount {
+	mounts := make([]specs.Mount, 0)
+
+	// if disable hostfiles, we will not mount the host files into container.
+	if c.Config.DisableHostsfiles {
+		return mounts
+	}
+
+	if c.HostnamePath != "" {
+		_, err := os.Stat(c.HostnamePath)
+		if err != nil {
+			logrus.Warnf("HostnamePath set to %v, but stat error: %v, skip it", c.HostnamePath, err)
+		} else {
+			mounts = append(mounts, specs.Mount{
+				Source:      c.HostnamePath,
+				Destination: "/etc/hostname",
+				Type:        "bind",
+				Options:     []string{"rbind", "rprivate"},
+			})
+		}
+	}
+
+	if c.HostsPath != "" {
+		_, err := os.Stat(c.HostsPath)
+		if err != nil {
+			logrus.Warnf("HostsPath set to %v, but stat error: %v, skip it", c.HostsPath, err)
+		} else {
+			mounts = append(mounts, specs.Mount{
+				Source:      c.HostsPath,
+				Destination: "/etc/hosts",
+				Type:        "bind",
+				Options:     []string{"rbind", "rprivate"},
+			})
+		}
+	}
+
+	if c.ResolvConfPath != "" {
+		_, err := os.Stat(c.ResolvConfPath)
+		if err != nil {
+			logrus.Warnf("ResolvConfPath set to %v, but stat error: %v, skip it", c.ResolvConfPath, err)
+		} else {
+			mounts = append(mounts, specs.Mount{
+				Source:      c.ResolvConfPath,
+				Destination: "/etc/resolv.conf",
+				Type:        "bind",
+				Options:     []string{"rbind", "rprivate"},
+			})
+		}
+
+	}
+
+	return mounts
+}
+
+// isSetupNetworkMount checks whether set network mount.
+func isSetupNetworkMount(mount *types.MountPoint, c *Container) bool {
+	if mount.Destination == "/etc/hostname" {
```

Comment:
How about code below ?
```
if utils.StringInSlice(mount.Destination, []string{"/etc/hostname", "/etc/hosts", "/etc/resolv.conf"})
    xxx
```

Reply:
the outer function will iterator all mountpoints